presto-main/src/main/java/com/facebook/presto/execution/StageExecutionState.java

```diff
@@ -61,6 +61,10 @@
      * was not in this stage.
      */
     ABORTED(true, true),
+    /**
+     * Query execution failed, but may retry.
+     */
+    TENTATIVE_FAILED(false, false),
```

Comment:
I agreed with @arhimondr  this state sounds a bit weird. What's the new state we are thinking? e.g. `PREPARING_FOR_RETRY`?

Reply:
The idea is to get rid of it entirely.  Let the stage actually fail and deal with the retry logic in the FAILED state change listener.  There was some reason I originally needed this non-done failed state, but I don't remember why, so I think it may have been an artifact of something that isn't there in the final version of this code.